### PR TITLE
#GS2-227 + GS2-228 Implement Pexels image search endpoint + karate tests

### PR DESCRIPTION
**Pull request by VitaliiChernenko**
## Summary
Implemented image search functionality using the Pexels API and exposed it through the Orders API management endpoint.

## What's Included
- Added `PexelsImageSearchClientAdapter` with improved error handling and null-safety.
- Integrated random page selection for more diverse image results.
- Added new management endpoint: `/v1/management/products/images/search`.
- Implemented full Karate test coverage for image search flow.
- Added URL validation and non-null checks in tests.
- Introduced constants and small refactorings for clarity and maintainability.

## Summary by CodeRabbit

* **New Features**
  * Product image search endpoint for managers to retrieve image candidates via external image service.
  * Configurable external image service settings (API key, base URL, page size).

* **Bug Fixes**
  * Graceful handling when image search service is unavailable, returning HTTP 503 with clear error message.

* **Tests**
  * Added unit and end-to-end tests covering success, pagination, edge cases, and error scenarios for image search.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: code/orders-api-rest-clients/pexels-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-api-rest-clients/pexels-api/pom.xml:1]: This module inherits the parent’s dependency tree, so the reported vulnerabilities come from the root pom.xml.
The Pexels feature itself does not introduce any new dependencies or vulnerabilities.
Parent POM upgrades can be handled separately as part of dependency management.

File: code/orders-api-rest-clients/pexels-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  <artifactId>pexels-api</artifactId>
+
+  <properties>
+    <maven.compiler.source>17</maven.compiler.source>
+    <maven.compiler.target>17</maven.compiler.target>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+</project>
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-api-rest-clients/pexels-api/pom.xml:18]: Lombok is already defined in the root POM under <dependencies> with scope=provided 
and is inherited by all modules, including pexels-api, through the Maven parent 
hierarchy. The annotation processor is also configured globally in the root 
maven-compiler-plugin. Adding Lombok locally to pexels-api would duplicate the 
dependency and is not necessary.

File: code/orders-api-rest/src/main/java/com/academy/orders/apirest/products/controller/ProductsManagementController.java [hunk 3]
```diff
@@ -104,4 +107,10 @@ public ResponseEntity<Integer> getCountOfDiscountedProducts() {
     Integer discountedProductCount = getCountOfDiscountedProductsUseCase.getCountOfDiscountedProducts();
     return ResponseEntity.ok(discountedProductCount);
   }
+
+  @Override
+  @PreAuthorize("hasAuthority('ROLE_MANAGER')")
+  public ResponseEntity<List<String>> searchProductImagesCandidates(String query) {
+    return ResponseEntity.ok(pexelsImageSearchUseCase.searchImages(query));
+  }
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-api-rest/src/main/java/com/academy/orders/apirest/products/controller/ProductsManagementController.java:115]: Thanks for the suggestion!  

In this case, explicit `@RequestParam` and validation annotations in the controller method are not strictly necessary because this controller implements the OpenAPI-generated `ProductsManagementApi` interface, which already defines:

```java
@NotNull
@Size(min=3, max=100)
@RequestParam(value = "query", required = true)
String query
```
Spring reads these annotations from the interface and enforces validation at runtime, which is why the application already rejects invalid requests (like query="ip").

There is no need to add the same annotations in the implementation.

File: code/orders-api-rest-clients/pexels-api/src/main/java/com/academy/orders/external/pexels/config/PexelsClientProperties.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  /**
+   * Base URL of Pexels API, e.g. "https://api.pexels.com/v1"
+   */
+  private String baseUrl;
+
+  /**
+   * Number of photos per request (default 8).
+   */
+  private Integer perPage = 8;
+}
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-api-rest-clients/pexels-api/src/main/java/com/academy/orders/external/pexels/config/PexelsClientProperties.java:33]: Thanks. Good suggestion.
Fixed.

File: code/orders-infrastructure/src/main/java/com/academy/orders/infrastructure/pexels/adapter/PexelsImageSearchClientAdapter.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    PexelsSearchResponse body = callPexels(url, request);
+    return body != null ? body.getTotalResults() : 0;
+  }
+
+  private int pickRandomPage(int totalResults) {
+    int perPage = properties.getPerPage();
+    int maxPages = (int) Math.ceil((double) totalResults / perPage);
+    return (maxPages > 1) ? random.nextInt(maxPages) + 1 : 1;
+  }
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-infrastructure/src/main/java/com/academy/orders/infrastructure/pexels/adapter/PexelsImageSearchClientAdapter.java:73]: Not necessary, because there are validations in the PexelsClientProperties. perPage can't be negative or zero.
Thanks.

File: code/orders-api-rest-clients/pexels-api/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-api-rest-clients/pexels-api/pom.xml:1]: There is no need to create a GitHub issue. Thanks.

File: code/orders-boot/src/main/java/com/academy/orders/boot/Application.java [hunk 6]
```diff
@@ -1,16 +1,15 @@
...
 import org.springframework.boot.actuate.autoconfigure.metrics.web.tomcat.TomcatMetricsAutoConfiguration;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
 import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication(exclude = {SystemMetricsAutoConfiguration.class, TomcatMetricsAutoConfiguration.class})
 @ComponentScan(basePackages = "com.academy.orders")
-@EnableConfigurationProperties(UsersConfig.class)
+@ConfigurationPropertiesScan
```
[the remark this answers is not in the document]
Comment (vcher4) [reply code/orders-boot/src/main/java/com/academy/orders/boot/Application.java]: Good catch.
Thanks.
Fixed.

**Review by vitkikh — APPROVED** [review]
